HelpMyStreetFE/HelpMyStreetFE/Services/AddressService.cs

```diff
@@ -184,5 +190,48 @@ public async Task<double> GetDistanceBetweenPostcodes(string postCode1, string p
                 }
             }, $"{CACHE_KEY_PREFIX}-postcode-distances-{postCode1}-{postCode2}", RefreshBehaviour.DontWaitForFreshData, cancellationToken);
         }
+
+        public async Task<double> GetDistanceFromPostcodeForCurrentUser(string postCode, CancellationToken cancellationToken)
+        {
+            var httpContext = _httpContextAccessor.HttpContext;
+            if (httpContext.User != null && httpContext.User.Identity.IsAuthenticated)
```

Comment:
This code is duplicated in the AuthService.  Please can we call the AuthService at this point?

Reply:
Apart from removing the duplication, that should make use of some user caching :)